cockatrice/src/update_checker.cpp

```diff
@@ -37,18 +34,33 @@ void UpdateChecker::check()
 #if defined(Q_OS_OSX)
 bool UpdateChecker::downloadMatchesCurrentOS(QVariant build)
 {
-   return build
+    return build
             .toMap()["name"]
             .toString()
-            .contains("osx");
+            .endsWith(".dmg");
 }
+
 #elif defined(Q_OS_WIN)
 bool UpdateChecker::downloadMatchesCurrentOS(QVariant build)
 {
-       return build
-            .toMap()["name"]
-            .toString()
-            .contains("exe");
+    QString wordSize = QSysInfo::buildAbi().split('-')[2];
+    QString arch;
+    if (wordSize == "llp64") {
+        arch = "win64";
+    } else if (wordSize == "ilp32") {
+        arch = "win32";
+    } else {
+        qWarning() << "Error checking for upgrade version: wordSize is" << wordSize;
+        return false;
+    }
+
+    auto fileName = build
+        .toMap()["name"]
+        .toString();
+    // Checking for .zip is a workaround for the May 6th 2016 release
+    auto zipName = arch + ".zip";
```

Comment:
Hmm?
We have the initial updater in Gatewatchers --> no update possible for sure because it checks for newer releases, which there are none at that time
Shadows release: people with Gatewatchers should've been able to update, but since bintray changed their api it wasn't working

We had the same updater code in Shadows, Zach then quickly resolved the bintray api change after realizing and testing the new official release (auto-update from gatewatchers was not possible) and re-released Shadows a second time shortly after with that fix.
In the same commit he also changed how checking works because we had a "qt5.exe" check inside, he just changed it to "exe" (https://github.com/Cockatrice/Cockatrice/pull/1974/files#diff-f3a13ff16ad53a26fc8911e00b6c543fL52)
Which itself leads to problems described by ctrlaltca for names like "Vort**exe**s", but also made the updater not work again on windows because it finds two exe files on bintray (which got separated before by qt4.exe and qt5.exe checks). So he introduced the zip workaround and changed one of the exe files on bintray to a zip file (containing the exe) to avoid the need of a code change again (and a 3rd Shadows release).

As far as I understand it, the only downside would be that windows users on Shadows release can just update properly to the upcoming next version as long as we keep the zip file as a workaround in bintray. I don't see the need for having the zip code in the app.
But please enlighten me. :)
There might be a problem at one point for some people which haven't updated for a longer time and try to jump from Shadows to the after next release once we go back to normal bintray hosting. But that's it.

They can update from shadows to the next release (which then has your fix in) and will safely bring them to the next release after as well - even with two exe files up again on bintray!

So we need to get your fix in (without zip code), release new version with just one exe on bintray, features features features, next release with two exe files on bintray. All good?!